### PR TITLE
Refactoring templates, cross-device sync.

### PR DESCRIPTION
**Pull request by funkyhippo**
Nobody:
Me: hey wouldn't it be nice if we had cross-device sync so we could easily switch between devices? 

This is a bit of a huge PR, I should've split it between the refactor (moving all the templates to a common, feature-controlled one) and the actual cross-device sync.

Refactor:
- Moved a bunch of the page data methods to reader views
- Deleted all the different templates for MD/NH/FS so it's shared now
- Tried my best to remove jQuery but bootstrap needs it
- Removed the referral templates

Cross-device sync:
- [x] Moved common proxyHistory handlers into a common history handler
- [x] Integrated [remotestorage](https://remotestorage.io/) to handle cross device sync
- [x] Add history pinning
- [x] Consolidate schema for series
- [x] Further modify remotestorage library to bypass IndexedDB (it's pretty buggy)
- [x] Fix the god-awful CSS for the login/sync status widget
- [x] Add instructions in tutorial card

The syncing doesn't require any dependencies on guya's side, and doesn't change any of the hosting requirements since it's a third-party service that'll handle the syncing. Make an account here: [5apps](https://5apps.com/storage/beta), and try it [here](https://dev.waterloo.engineering/). To get the login widget, click the twintail guya 8 times and it'll show up on the navbar at the top right.

I'm hosting it temporarily to test since it's a pretty huge change.

### CODE REVIEW

File: reader/templates/reader/reader.html [hunk 1]
```diff
@@ -24,35 +24,15 @@
...
-		{% include "proxy_history.html" %}
+		{% include "history.html" %}
 	</head>
 	
 <body>
-	<!-- Google Tag Manager (noscript) -->
-	<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-WWRXZB4"
-	height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
-	<!-- End Google Tag Manager (noscript) -->
+	{% include "tracking.html" %}
```
Comment (appu1232) [inline reader/templates/reader/reader.html:35]: Making this a template is nice. I think we should do the same thing with the meta tags throughout the other pages too (not you or in this pr, just a good note to have).

Comment (Algoinde) [reply reader/templates/reader/reader.html:35]: How do we go about overloading the template, tho? When og:desc modifications are needed

Comment (appu1232) [reply reader/templates/reader/reader.html:35]: Template tags would do the trick no?

Comment (appu1232) [reply reader/templates/reader/reader.html:35]: We shouldn't do any hardcoding anyway. Ideally all business logic should be determined and sent from the backend.

**Comment by funkyhippo** [comment]
I finally got off my lazy ass to finish this. Feel free to try it out [here](https://dev.waterloo.engineering/).

**Comment by Algoinde** [comment]
On mobile, the sync config button cannot be reached in any way.

**Pull request merged**